src/placeos-triggers/loader/trigger.cr

```diff
@@ -0,0 +1,28 @@
+require "placeos-models/trigger"
+require "placeos-resource"
+
+module PlaceOS::Triggers::Loader
+  class Trigger < Resource(Model::Trigger)
+    getter mapping : Mapping
+
+    def initialize(@mapping, **args)
+      super(**args)
+    end
+
+    def process_resource(action : Resource::Action, resource trigger : Model::Trigger) : Resource::Result
+      trigger_id = trigger.id.as(String)
```

Comment:
Casting like this feels... odd. A `#not_nil!` is marginally better as it stops the actual type from leaking into implementations here. Neater, but more involved solution is bubbling this up to `rethink-orm` to provide a non-nillable accessor.

Reply:
I've got that on a local branch, I'll put together a PR

Reply:
Will mark this as resolved for now